### ISSUE TITLE
Simulate.mouseEnter and Simulate.mouseLeave not working

### ISSUE THREAD

**Issue by leoasis** [facebook/react#1297]
While testing I wanted to simulate `mouseEnter` and `mouseLeave` events, but for some reason they are not firing the handlers in the component.

Here's a fiddle to show you: http://jsfiddle.net/leoasis/RH2vm/
You'll see that with the Simulate functions, only the click is actually triggered, while the other ones not. With the real events everything works perfectly.

I tried to debug a little bit what's going on, but didn't get much information, especially since it's the first time I'm diving into the events mechanism internals in React.

If you want more info, then I can try and dig a bit deeper, but I think the fiddle is clear enough.

**Comment by warmhug** [comment]
Do we have a good solution for such case now? or It can not be solved?

**Comment by ryanzec** [comment]
I am manually applying this patch until it is pulled into the core:

https://github.com/facebook/react/pull/1366

**Comment by ryanzec** [comment]
Any update on this?

**Comment by brianblocker** [comment]
I was able to get my tests to pass using SimulateNative. Here's my codez:

```javascript
// and_or_component.jsx
  render: function () {
    var props = {
      className:    'chain and-or',
      onMouseLeave: this._handleMouseLeave,
      onClick:      this._toggleEditing
    };

    return (
      <span {...props}>
        Some content here
      </span>
    );
  },
  _handleMouseLeave: function () {
    this.setState({editing: false});
  },

// and_or_component.jsx.test
  it('sets state to editing:false when the mouse leaves', function () {
    var and_or  = React.TestUtils.renderIntoDocument(<AndOr />);
    var node    = and_or.getDOMNode();

    TestUtils.Simulate.click(node);
    TestUtils.SimulateNative.mouseOut(node);

    expect(and_or.state.editing).to.equal(false);
  });
```

**Comment by ryanzec** [comment]
@brianblocker I tried that using both `mouseEnter` and `mouseOver` instead of `mouseOut` but it still did not trigger my `mouseEnter` event.

**Comment by phoenixbox** [comment]
@ryanzec
If it helps - the difference which worked for me was calling `SimulateNative` with a `DOMNode` and not a `ReactNode`. Pass & fail states listed along with full test case below:

##### Passed
TestUtils.SimulateNative.mouseOver(chosenListItem.getDOMNode());

##### Failed
TestUtils.SimulateNative.mouseOver(chosenListItem);


```
it.only('onMouseOver of an item sets its class to focused', function () {
        let listItems = TestUtils.scryRenderedComponentsWithType(this.component, ListGroupItem);
        let chosenListItem = listItems[0];

        TestUtils.SimulateNative.mouseOver(chosenListItem.getDOMNode());

        assert.isTrue(chosenListItem.getDOMNode().classList.contains('tag-item__focused'));
});
```

**Comment by Kureev** [comment]
No update so far? Maybe some workarounds?

**Comment by spicyj** [comment]
@Kureev This was fixed in #1366 in April as you can see a few messages up; it'll be in the 0.14 release. If you search "workaround" on this page you'll see that I also posted a workaround over a year ago.

**Comment by Kureev** [comment]
Sorry, really missed it. Thanks for it!
